addons/point_of_sale/models/pos_config.py

```diff
@@ -729,7 +729,15 @@ def _get_available_product_domain(self):
             ('sale_ok', '=', True),
         ]
         if self.limit_categories and self.iface_available_categ_ids:
-            domain.append(('pos_categ_ids', 'in', self.iface_available_categ_ids.ids))
+            pos_combos = self.env['pos.combo'].search([('combo_line_ids', '!=', False)])
+            # Retrieve all combo products, combine available and combo category IDs, and apply to domain
+            combo_product_ids = pos_combos.mapped('combo_line_ids.product_id')
```

Comment:
```suggestion
            combo_product_ids = pos_combos.combo_line_ids.product_id
```

What if we directly use `pos_combos.combo_line_ids.product_id` ?